src/components/date-picker/README.md

```diff
@@ -0,0 +1,52 @@
+### JavaScript
+
+The date picker is built on top of [Flatpickr](https://chmln.github.io/flatpickr/), so the [events](https://chmln.github.io/flatpickr/events/) and [config options](https://chmln.github.io/flatpickr/options/) that comes with Flatpickr is therefore available to the date picker by using the reference ``. This includes methods for setting a min date, max date, disabling date(s), specifiying a range of dates, and more.
+
+#### Public Methods
+
+| Name    | Params  | Description          |
+|---------|---------|----------------------|
+| release |         | Deletes the instance |
+
+#### Options
+
+| Option                      | Default Selector                    | Description                                           |
+|-----------------------------|-------------------------------------|-------------------------------------------------------|
+| selectorInit                | [data-date-picker]                  | Element for initializing instance                     |
+| selectorDatePickerInput     | [data-date-picker-input]            | Input element                                         |
+| selectorDatePickerInputFrom | [data-date-picker-input-from]       | For ranged calendars only: The "From date" input element  |
+| selectorDatePickerInputTo | [data-date-picker-input-to]       | For ranged calendars only: The "To date" input element  |
+| selectorDatePickerIcon | [data-date-picker-icon]       | Calendar icon  |
+| classCalendarContainer | .bx--date-picker__calendar       | Class selector for the calendar container  |
+| classMonth | .bx--date-picker__month       | Class selector for the calendar month  |
+| classWeekdays | .bx--date-picker__weekdays       | Class selector for the calendar weekdays  |
+| classDays | .bx--date-picker__days       | Class selector for the calendar days  |
+| classWeekday | .bx--date-picker__weekday      | Class selector for a calendar weekday  |
+| classDay | .bx--date-picker__day       | Class selector for a calendar day  |
+| attribType | data-date-picker-type       | Specifies the calendar mode (single or range) |
+| dateFormat | 'm/d/Y' | The date format given to the calendar instance |
+
+
+### FAQ 
+
+#### Using and understanding Date Picker
+
+There are 3 different date picker types:
+
+| Type | data-date-picker-type |
+|--|--|
+| A simple date picker without a calendar | No data attributes needed |
+| A single date picker | [data-date-picker-type="single"] |
+| A ranged date picker | [data-date-picker-type="range"] |
+
+**The simple date picker** is a text input without a calendar. You can specify the pattern for the text input to make sure the user enters the correct date format. The default pattern that ships with the simple date picker is `\d{1,2}/\d{4}` (for short date pickers) and `\d{1,2}/\d{1,2}/\d{4}`. The simple date picker does not require any JavaScript.
```

Comment:
Can you also include the date format that the regex pattern expect? 

For example, docs can be edited to say:
The default regex pattern that ships with
simple date picker is `\d{1,2}/\d{4}` ('dd/yyyy' for short date pickers) and `\d{1,2}/\d{1,2}/\d{4}` ('dd/mm/yyyy' or mm/dd/yyyy)